plexapi/alert.py

```diff
@@ -52,8 +56,13 @@ def run(self):
         # create the websocket connection
         url = self._server.url(self.key, includeToken=True).replace('http', 'ws')
         log.info('Starting AlertListener: %s', url)
-        self._ws = websocket.WebSocketApp(url, on_message=self._onMessage,
-                                          on_error=self._onError)
+
+        if self._socket:
+            self._ws = websocket.create_connection(url=url, on_message=self._onMessage, on_error=self._onError,
+                                                   socket=self._socket)
+        else:
+            self._ws = websocket.WebSocketApp(url, on_message=self._onMessage, on_error=self._onError)
```

Comment:
`WebSocketApp` accepts a `socket` argument.

https://websocket-client.readthedocs.io/en/latest/app.html#websocket._app.WebSocketApp

Reply:
It doesn't in earlier versions. I assumed you wanted to keep compatibility with older versions based on this comment, below.

```py
    def _onMessage(self, *args):
        """ Called when websocket message is received.
            In earlier releases, websocket-client returned a tuple of two parameters: a websocket.app.WebSocketApp
            object and the message as a STR. Current releases appear to only return the message.
            We are assuming the last argument in the tuple is the message.
            This is to support compatibility with current and previous releases of websocket-client.
        """
```

I can update it if you prefer though?

Reply:
`create_connection` returns a `WebSocket` object, so it's not quite the same as `WebSocketApp` object.

I think it's fine to target the latest version of `websocket-client`. They actually reverted back to returning a two-tuple in the latest version so the docstring for `_onMessage` and `_onError` is wrong anyways. I think the entire comment about previous versions of `websocket-client` can be removed.

Reply:
To avoid a bunch if if/else logic, I added `websocket-client>=1.3.3` to the `requirements.txt`. It was also kind of annoying to have to install it separately. I think there's a way to set optional dependencies with the setup.py, but I'm not sure how... or how to have pip install them.